src/Symfony/Component/Validator/Constraints/AllValidator.php

```diff
@@ -44,12 +44,12 @@ public function validate($value, Constraint $constraint)
             $validator = $context->getValidator()->inContext($context);
 
             foreach ($value as $key => $element) {
-                $validator->atPath('['.$key.']')->validate($element, $constraint->constraints);
+                $validator->atPath('['.$key.']')->validate($element, $constraint->constraints, $constraint->groups);
```

Comment:
I asked the same in #17696 too: Don't we have to use `$context->getGroup()` here to not accidentally widen the validation scope?

Reply:
I think getters is better than properties, but there is `$constraint->constraints` not `$constraint->getConstraints()`. So I wrote the same way.
Should I change only groups, change both or no one?

Reply:
No, what I mean is fetching the groups from the context but not from the constraint? There may be more groups configured on the constraint than what groups are actually used to validate.

Reply:
Nope, It's not the same. Please see my original issue: https://github.com/symfony/symfony/issues/23479
I have attached xdebug screenshots there.
Also I have post a link to my symfony standart fork with easy code, which is broken with current version of Symfony, but it works with my fixes. You can clone it and make sure that `$context->getGroup` will have only a `Default` string, while `$constraint->constraints` will have an array of needed groups: `["info", "Default", "Category]` for the `Category` entity as it was set in [validation.yml](https://github.com/Aliance/SymfonyTest/blob/86404b6899300aa6d74e7a7d1ee471c76cffa131/src/AppBundle/Resources/config/validation.yml#L17-L17)

Reply:
You are right that there is still a problem. But just passing the constraints here does not work either. For example, if you take your example project and modify the controller to only run validation in the `Default` and `modify` group, the `Length` constraint will still be evaluated which is only part of the `info` group.